src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -141,5 +134,31 @@ def _produce_billing_outcome(
             quantity=quantity,
         )
 
+    def _flag_metric_received_for_project(self, generic_metric: GenericMetric) -> None:
+        project_id = generic_metric["project_id"]
+
+        # We try to extract the MRI from the metric_id since our goal is to check whether the MRI belongs to a metric
+        # in the `custom` namespace.
+        metric_mri = self._resolve(generic_metric["mapping_meta"], generic_metric["metric_id"])
+        parsed_mri = parse_mri(metric_mri)
+        if parsed_mri is None or not is_custom_metric(parsed_mri):
+            return
+
+        try:
+            project = Project.objects.get_from_cache(id=project_id)
```

Comment:
this is fairly unoptimized. it's possible that the consumer will backlog. i suggest to do the rollout with somebody from ops present to scale up consumer if needed on a whim